packages/analytics/src/client/client.ts

```diff
@@ -0,0 +1,121 @@
+import { AnalyticsClientConfig, GenericAnalyticsClient, JsonMap } from '../utils/types.shared';
+
+export type AnalyticsClientOptions = Pick<
+  AnalyticsClientConfig,
+  'defaultProperties' | 'defaultTraits'
+>;
+
+export class AnalyticsClient<T extends GenericAnalyticsClient> {
+  constructor(
+    private readonly client: T,
+    private readonly options: AnalyticsClientOptions
+  ) {
+    this.client = client;
+    this.options = options;
+  }
+
+  private async track(event: string, properties?: JsonMap) {
+    return this.client.track(event, { ...properties, ...this.options.defaultProperties });
+  }
+
+  public async screen(name: string, properties?: JsonMap) {
+    return this.client.screen(name, { ...properties, ...this.options.defaultProperties });
+  }
+
+  public async group(groupId: string, traits?: JsonMap) {
+    return this.client.group(groupId, { ...traits, ...this.options.defaultTraits });
+  }
+
+  public async identify(userId: string, traits?: JsonMap) {
+    return this.client.identify(userId, { ...traits, ...this.options.defaultTraits });
+  }
+
+  private errorToJsonMap(error: Error) {
+    return {
+      errorMessage: error.message,
+      errorName: error.name,
+    };
+  }
+
+  // TODO: replace view_transaction_confirmation with this method call.
+  public async trackViewTransactionConfirmation(properties: { symbol: string }) {
+    return this.track('Transaction Confirmation Viewed', properties);
```

Comment:
Not sure about the naming pattern change here? @markmhendrickson is this change ok? I know we coordinated the `snake_case` for naming quite awhile ago.

Reply:
I'm cool with updating the naming pattern. Let's just be sure to alias old names to the new ones if we update them, so we can continue reporting consistently.

Reply:
Mentioned on slack, but question why we'd make this change? What's so superior about this naming scheme that warrants the effort of doing the aliasing? [Per Mixpanel docs](https://docs.mixpanel.com/docs/data-structure/events-and-properties#name-events-and-properties-consistently:~:text=Generally%2C%20adopting%20snake_case%20for%20your%20event%20and%20property%20names%20tend%20to%20be%20more%20robust%2C%20especially%20if%20you%20plan%20to%20export%20your%20Mixpanel%20data%20to%20downstream%20processes%20such%20as%20data%20warehouses.)

> Generally, adopting snake_case for your event and property names tend to be more robust, especially if you plan to export your Mixpanel data to downstream processes such as data warehouses.